source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+/*
+  Copyright (C) 2011-2021 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/simulator.h>
+#include <aspect/utilities.h>
+#include <aspect/postprocess/plateness_statistics.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/spherical_shell.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    PlatenessStatistics<dim>::execute (TableHandler &statistics)
+    {
+      // Create a quadrature formula based on the velocity element.
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      const unsigned int quadrature_degree = this->introspection().polynomial_degree.temperature;
+      const QGauss<2> quadrature_formula_face(quadrature_degree);
+
+      FEFaceValues<3> fe_face_values (this->get_mapping(),
+                                      this->get_fe(),
+                                      quadrature_formula_face,
+                                      update_values |
+                                      update_quadrature_points |
+                                      update_JxW_values);
+
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      std::vector<double> local_second_invariant_of_strain_rate_integral;
+      std::vector<double> local_surface_area_integral;
+      std::vector<double> local_second_invariant_of_strain_rate_and_area;
+      std::vector<double> sorted_local_second_invariant_of_strain_rate;
```

Comment:
You need to rethink which of these vectors you need, and which ones actually need to be vectors and which ones are only double. Also the name `local_second_invariant_of_strain_rate_and_area` is confusing, a single vector of numbers can not contain two different things (strain rate and area)

Reply:
Yes, thanks. I have now set "local_second_invariant_of_strain_rate_integral" and "local_surface_area_integral" to be purely doubles. They are the second invariant of strain rate over the whole surface and total area, respectively. But I now save each local value into the vectors: "local_second_invariant_of_strain_rate[q]" and "local_surface_area[q]" so prepare for the cumulative addition loop for reaching f_80. This "local_second_invariant_of_strain_rate_and_corresponding_area" should contain "local_second_invariant_of_strain_rate" in column 1 and "local_surface_area" in column 2 so that we can re-order the each row based on the size in column 1.